### DISCUSSION TITLE
How does it work order value in @ChangeUnit annotation

### DISCUSSION THREAD

**Discussion by ahamdi-iungo** [flamingock/flamingock-java#696] [Q&A]
I accidentally defined two @ChangeUnit classes with the same order and mongock executed them correctly. But I would like to understand how it works. Did he do them randomly? Can it be prevented with one mistake?

**Comment by osantana85** [comment]
Hello @ahamdi-iungo , when two or more ChangeUnits have the same order, Mongock will use the class canonical name to compare them.

Hope this helps you.

**Reply by ahamdi-iungo** [reply]
thank you so much! 
I have another question. I tried define two classes annotated with @ChangeUnit with the same id and they were done correctly. I was expecting an error. Isn't the id unique?

**Reply by osantana85** [reply]
There is an open issue related to this [here](https://github.com/mongock/mongock/issues/680).

We probably will change it in a future major version and will document it properly.

**Reply by ahamdi-iungo** [reply]
thank you @osantana85 i appreciate your answers
